### PR TITLE
Bug 2054254: Rearrange test cases and gather more debugging information

### PR DESCRIPTION
**Pull request by slaskawi**
https://bugzilla.redhat.com/show_bug.cgi?id=2054254

This Pull Request doesn't provide a fix for the BZ mentioned the above. It provides a bit more verbose information that might be helpful for further debugging.

Furthermore, this Pull Request changes the order of test cases. This way, we should know whether this is a certificate related problem or a token one.

### CODE REVIEW

**Comment by deads2k** [comment]
doesn't build

**Comment by slaskawi** [comment]
Shuld be better now.

File: test/extended/oauth/requestheaders.go [hunk 1]
```diff
@@ -234,6 +238,19 @@ var _ = g.Describe("[Serial] [sig-auth][Feature:OAuthServer] [RequestHeaders] [I
 	})
 })
 
+func gatherPostMortem(oc *exutil.CLI) {
+	authn, err := oc.AdminConfigClient().ConfigV1().ClusterOperators().Get(context.Background(), "authentication", metav1.GetOptions{})
+	if err != nil {
+		e2e.Logf("Error getting authentication operator: %v\n", err)
+	}
+	e2e.Logf("Authentication %v\n", authn)
+	routerCA, err := oc.AdminKubeClient().CoreV1().ConfigMaps("openshift-config-managed").Get(context.Background(), "default-ingress-cert", metav1.GetOptions{})
```
Comment (stlaz) [inline test/extended/oauth/requestheaders.go:247]: I'm not sure how retrieving this certificate helps, it's more or less a constant and if it were broken, many other things would be broken as well.

Maybe you may want to check:
a) the actual status of the oauth-openshift deployment
b) current config in the openshift-authentication NS
c) the logs of all the oauth-openshift deployment pods

File: test/extended/oauth/requestheaders.go [hunk 2]
```diff
@@ -211,16 +211,20 @@ var _ = g.Describe("[Serial] [sig-auth][Feature:OAuthServer] [RequestHeaders] [I
 				respDump, err := httputil.DumpResponse(resp, true)
 				o.Expect(err).NotTo(o.HaveOccurred())
 				if len(tc.expectedError) == 0 && resp.StatusCode != 200 && resp.StatusCode != 302 {
+					gatherPostMortem(oc)
```
Comment (stlaz) [inline test/extended/oauth/requestheaders.go:214]: Don't ginkgo or gomega provide a better mechanism to perform actions upon a failure?

Comment (ibihim) [reply test/extended/oauth/requestheaders.go:214]: Could you be more explicit? You mean adding it to some `afterEach` function or do you mean that there exists already a `gatherPostMortem` in our test suites?

**Pull request closed without merging**